src/main/java/com/google/devtools/build/lib/rules/genrule/GenRuleBase.java

```diff
@@ -321,6 +326,31 @@ public String lookupVariable(String variableName) throws ExpansionException {
         return expandSingletonArtifact(filesToBuild, "$@", "output file");
       }
 
+      if (variableName.equals("OUT_ROOT_DIR")) {
+        // The output root directory. This variable expands to the package's root directory
+        // in the genfiles tree.
+        Path dirPath;
+        PathFragment dirFragment;
+        if (ruleContext.getRule().hasBinaryOutput()) {
```

Comment:
use `RuleContext#getBinOrGenfilesDirectory()` instead. Has the exact same logic.

Reply:
ping?

Reply:
I removed the part where I needed the Directory. There isn't an equivalent method for dirFragment afaics.